department-of-veterans-affairs/abd-vro
# svc-bip-api: Add externalKey and externalUserId to JWT

## What was the problem?

BIP Claims API is upgrading to a new version that will require all requests to have the claims `externalUserId` and `externalKey` in the JWT. 

Associated tickets or Slack threads:
- #3448 
- #3462 

## How does this fix it?[^1]

* Adds the `externalUserId` and `externalKey` fields to the base request model BipPayloadRequest.java, and changes it from a interface to a class so that all the requests models can inherit these new fields. The fields are `WRITE_ONLY` meaning that they are deserialized from strings into the POJO, but not serialized from POJO into a string. This means that requests POJOs from the RabbitMQ will have the fields, but they will not be sent in the request to the downstream BIP Claims API.

## How to test this PR
- unit tests
- integration tests
- passing CI

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
